feature/groups/src/main/java/com/mifos/feature/groupsList/presentation/GroupsListScreen.kt

```diff
@@ -0,0 +1,383 @@
+@Composable
+fun GroupsListRoute(
+    onAddGroupClick: () -> Unit,
+    onGroupClick: (Group) -> Unit,
+    onSyncClick: (List<Group>) -> Unit,
+    modifier: Modifier = Modifier,
+    viewModel: GroupsListViewModel = hiltViewModel()
+) {
+    val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+
+    val isRefreshing by viewModel.isLoading.collectAsStateWithLifecycle()
+    val swipeRefreshState = rememberSwipeRefreshState(isRefreshing = isRefreshing)
+
+    GroupsListScreen(
+        modifier = modifier,
+        uiState = uiState,
+        swipeRefreshState = swipeRefreshState,
+        onRefresh = viewModel::refreshData,
+        onAddGroupClick = onAddGroupClick,
+        onGroupClick = onGroupClick,
+        onSyncClick = onSyncClick,
+    )
+}
+
+
+@Composable
+fun GroupsListScreen(
+    modifier: Modifier = Modifier,
+    uiState: GroupsListState,
+    swipeRefreshState: SwipeRefreshState,
+    onRefresh: () -> Unit,
+    onAddGroupClick: () -> Unit,
+    onGroupClick: (Group) -> Unit,
+    onSyncClick: (List<Group>) -> Unit,
+) {
+    val lazyListState = rememberLazyListState()
+    val selectedItems = remember {
+        mutableStateListOf<Group>()
+    }
+
+    BackHandler(
+        enabled = selectedItems.isNotEmpty()
+    ) {
+        selectedItems.clear()
+    }
+
+    Scaffold(
+        modifier = modifier
+            .fillMaxSize(),
+        floatingActionButton = {
+            MifosFAB(icon = Icons.Default.Add, onClick = onAddGroupClick)
+        },
+        floatingActionButtonPosition = FabPosition.End,
+        topBar = {
+            AnimatedVisibility(
+                visible = selectedItems.isNotEmpty(),
+                enter = fadeIn(tween(500)),
+                exit = fadeOut(tween(500))
+            ) {
+                SelectionModeTopAppBar(
+                    itemCount = selectedItems.size,
+                    syncClicked = { onSyncClick(selectedItems.toList()) },
+                    resetSelectionMode = {
+                        selectedItems.clear()
+                    }
+                )
+            }
+        },
+    ) {
+        SwipeRefresh(
+            state = swipeRefreshState,
+            onRefresh = onRefresh,
+        ) {
+            Column(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(it)
+                    .padding(PaddingSmall)
+            ) {
+                Crossfade(
+                    targetState = uiState,
+                    label = "GroupsList::State",
+                ) { state ->
+                    when (state) {
+                        is GroupsListState.Empty -> {}
+
+                        is GroupsListState.Error -> {
+                            MifosSweetError(
+                                message = state.message,
+                                onclick = onRefresh,
+                            )
+                        }
+
+                        is GroupsListState.GroupsFromAPI -> {
```

Comment:
This approach is wrong, UI should not know from where data is coming should independent, make only one 
 GroupsListState.ShowUI

Reply:
In the ViewModel, it checks the user status to determine whether user is online. If the user is online, data will be fetched from the API. Otherwise, data will be retrieved from the local database.

Reply:
And, yes the approach might be wrong, and UI Layer should be independent